Terminal.Gui/Application/Application.Run.cs

```diff
@@ -186,7 +186,14 @@ public static RunState Begin (Toplevel toplevel)
 
         toplevel.LayoutSubviews ();
         toplevel.PositionToplevels ();
-        toplevel.FocusFirst (null);
+
+        // TODO: Should this use FindDeepestFocusableView instead?
+        // Try to set initial focus to any TabStop
+        if (!toplevel.AdvanceFocus (NavigationDirection.Forward, TabBehavior.TabStop))
```

Comment:
This feels like something that belongs in a different type, as it's making a specific behavior dictated by another class into something that this class is directly making decisions on. 

Alternatively, can't the AdvanceFocus method make this determination, potentially even based on type checks to avoid reliance on the enum? At minimum, the same operation, but performed by AdvanceFocus is a more appropriate place than here.

Reply:
@dodexahedron: Your writing is insightful, but the use of the pronoun “This” is frequently ambiguous and makes the reader work too hard to decode. 

For example:

“This feels like something that belongs in a different type, as it’s making a specific behavior dictated by another class into something that this class is directly making decisions on.”

To improve clarity, replace “This” with a specific noun. For instance:

“The behavior dictated by `View` should not be directly decided by `Application`.”

(I actually don't know if the above represents what you meant, BTW).

This change will make your references clearer and your argument stronger and will make your readers lives easier.

Reply:
Sorry, I know that still slips in quite a bit. I'm working on that per your previous request, though, so it ain't falling on deaf ears (blind eyes?).

Specifically, this is referring to the code block it annotates, which is that if/else construct that calls the same method (AdvanceFocus) on another object (twice) to take action on that same object. The entire operation is therefore something that `View` does, _to_ `View`, and thus belongs as a method in `View` itself. The AdvanceFocus method itself can still be what it is, but the if/else in this block should be either part of the AdvanceFocus method itself or another method in `View` that calls `AdvanceFocus`.

As part of Application, it's just tight coupling and violation of encapsulation.

Reply:
But also, _why can't you just read my mind????_ 😅

Reply:
> Sorry, I know that still slips in quite a bit. I'm working on that per your previous request, though, so it ain't falling on deaf ears (blind eyes?).
> 
> Specifically, this is referring to the code block it annotates, which is that if/else construct that calls the same method (AdvanceFocus) on another object (twice) to take action on that same object. The entire operation is therefore something that `View` does, _to_ `View`, and thus belongs as a method in `View` itself. The AdvanceFocus method itself can still be what it is, but the if/else in this block should be either part of the AdvanceFocus method itself or another method in `View` that calls `AdvanceFocus`.
> 
> As part of Application, it's just tight coupling and violation of encapsulation.

Great point. 

Turns out I was being stoopid^2 and all that really was needed here was a call to `top.SetFocus` anyway.

Reply:
> Turns out I was being stoopid^2 and all that really was needed here was a call to `top.SetFocus` anyway.

Exactly where I hoped that would end up once you saw it with fresh eyes. 😉

Hmmm... Instead of "stoopid," I kinda like the sound of "exploring alternative avenues in order to nurture nascent ingenuity," don't you? Much more enterprisey. 😆